### ISSUE TITLE
[Paper Note] Don't be lazy: CompleteP enables compute-efficient deep transformers, Nolan Dey+, NeurIPS'25, 2025.05

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#5695]
# URL
- https://arxiv.org/abs/2505.01618
# Authors
  - Nolan Dey
  - Bin Claire Zhang
  - Lorenzo Noci
  - Mufan Li
  - Blake Bordelon
  - Shane Bergsma
  - Cengiz Pehlevan
  - Boris Hanin
  - Joel Hestness
# Abstract
  - We study compute efficiency of LLM training when using different parameterizations, i.e., rules for adjusting model and optimizer hyperparameters (HPs) as model size changes. Some parameterizations fail to transfer optimal base HPs (such as learning rate) across changes in model depth, requiring practitioners to either re-tune these HPs as they scale up (expensive), or accept sub-optimal training when re-tuning is prohibitive. Even when they achieve HP transfer, we develop theory to show parameterizations may still exist in the lazy learning regime where layers learn only features close to their linearization, preventing effective use of depth and nonlinearity. Finally, we identify and adopt the parameterization we call CompleteP that achieves both depth-wise HP transfer and non-lazy learning in all layers. CompleteP enables a wider range of model width/depth ratios to remain compute-efficient, unlocking shapes better suited for different hardware settings and operational contexts. Moreover, CompleteP enables 12-34% compute efficiency improvements over the prior state-of-the-art. All experiments were run on Cerebras CS-3 systems. A minimal implementation is available at https://github.com/EleutherAI/nanoGPT-mup/tree/completep.
# Translation (by gpt-5-nano)
- 本研究では、モデルサイズが変化する際にモデルとオプティマイザのハイパーパラメータ（HPs）を調整するルール、すなわち異なるパラメータ化を用いたLLM訓練の計算効率を検討する。モデル深さの変化に跨って基礎HP（例：学習率）を最適値へ転送できないパラメータ化があり、実務者はスケールアップ時にこれらHPを再調整する（費用が高い）か、再調整が難しい場合には訓練を最適でないものとして受け入れる必要がある。HP転送を達成しても、パラメータ化は怠惰な学習領域に存在し得ることを理論的に示し、層が線形化に近い特徴のみを学習するため深さと非線形性を効果的に活用できなくなる。最後に、深さ方向のHP転送と全層での非怠惰な学習を同時に実現する CompleteP というパラメータ化を特定し採用する。CompleteP は、幅と深さの比の取りうる範囲をより広く保つことを可能にし、異なるハードウェア設定や運用環境に適した構成を解き放つ。さらに、CompleteP は従来の最先端と比較して12-34%の計算効率の改善を実現する。すべての実験は Cerebras CS-3 システム上で実行された。最小実装は以下のURLで入手可能です：https://github.com/EleutherAI/nanoGPT-mup/tree/completep
# Summary (by gpt-4o-mini)
- モデルサイズによるハイパーパラメータ調整のルールを検討し、従来のパラメータ化の限界を指摘。特に、最適な学習率の転送が不可能な場合には、再調整が高コストで難しいとされる。さらに、深さと非線形性の利点を活かせない怠惰な学習領域の問題を理論的に示す。新たに提案するCompletePパラメータ化は、深さと広さの比の範囲を拡大し、より効率的な訓練を実現。CompletePは、従来と比較して12-34%の計算効率の改善を示した。全ての実験はCerebras CS-3システム上で行われ、最小実装は提供されている。

**Comment by AkihikoWatanabe** [comment]
openreview:https://openreview.net/forum?id=lMU2kaMANl

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/eliebakouch/status/2061862867553382439?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
muP:
- #2583
